src/liballoc/arc.rs

```diff
@@ -169,7 +169,7 @@ impl<T> Arc<T> {
             weak: atomic::AtomicUsize::new(1),
             data: data,
         };
-        Arc { _ptr: unsafe { NonZero::new(mem::transmute(x)) } }
+        Arc { _ptr: unsafe { NonZero::new(mem::transmute(x)).unwrap() } }
```

Comment:
Since `box` is guaranteed to return a non-null ptr, couldn't this just be `mem::transmute(x)`?